Sources/ScipioKit/Resolver/PackageResolver.swift

```diff
@@ -0,0 +1,544 @@
+        )
+        allModules.insert(module)
+        return module
+    }
+
+    /// Normalizes an optional PackageCondition into an array.
+    private func normalizeConditions(_ condition: PackageCondition?) -> [PackageCondition] {
+        return condition.map { [$0] } ?? []
+    }
+
+    /// Determines the type of a module (e.g., Swift, Clang, binary) for a given target.
+    private func resolveModuleType(
+        of target: Target,
+        dependencyPackage: DependencyPackage
+    ) -> ResolvedModuleType {
+        if let cachedModuleType = cachedModuleType[target] {
+            return cachedModuleType
+        }
+
+        let resolvedModuleType = ModuleTypeResolver(
+            fileSystem: fileSystem,
+            rootPackageDirectory: packageDirectory,
+            target: target,
+            dependencyPackage: dependencyPackage
+        ).resolve()
+
+        cachedModuleType[target] = resolvedModuleType
+
+        return resolvedModuleType
+    }
+
+    private func resolveDependencyPackage(for manifest: Manifest) -> DependencyPackage {
+        guard let dependencyPackage = dependencyPackagesByName[manifest.name] else {
+            fatalError("Manifest \(manifest.name) refers to unknown package")
+        }
+        return dependencyPackage
+    }
+
+    private func loadManifest(for dependencyPackage: DependencyPackage) async throws -> Manifest {
+        if let cachedManifest = cachedDependencyManifests[dependencyPackage] {
+            return cachedManifest
+        } else {
+            let manifest = try await manifestLoader.loadManifest(for: dependencyPackage)
+            setActualPackageKinds(for: manifest)
+            cachedDependencyManifests[dependencyPackage] = manifest
+            return manifest
+        }
+    }
+
+    private func setActualPackageKinds(for manifest: Manifest) {
+        let actualPackageKinds = manifest.dependencies.reduce(into: [String: PackageKind]()) { partialResult, dependency in
+            switch dependency {
+            case .fileSystem(let fileSystem):
+                partialResult.updateValue(.fileSystem(fileSystem.path), forKey: fileSystem.identity)
+            case .registry(let registry):
+                partialResult.updateValue(.registry(registry.identity), forKey: registry.identity)
+            case .sourceControl(let sourceControl):
+                switch sourceControl.location {
+                case .local(let localURL):
+                    partialResult.updateValue(.localSourceControl(localURL), forKey: sourceControl.identity)
+                case .remote:
+                    partialResult.updateValue(.remoteSourceControl(""), forKey: sourceControl.identity)
+                }
+            }
+        }
+        self.resolvedPackageKinds.merge(actualPackageKinds) { _, new in new }
+    }
+
+    private func shouldBuild(_ target: Target.TargetKind) -> Bool {
+        target == .regular || target == .binary
+    }
+}
+
+private struct ModuleTypeResolver {
```

Comment:
This file is too-large so splitting into smaller chunks by extracting each component into its own files will be better (it would be okay to do that refactoring in a separate PR). Then it will be easier to test each component.

- ModuleTypeResolver
- PackageResolveExecutor
- ShowDependenciesParser

The types will be nested in PackageResolver.

```swift
extension PackageResolver {
    struct ModuleTypeResolver {
        ...
    }
}
```